File: Rasha_Toine/Rasha/QLForm/src/ast/literal/StringLiteral.java
package ast.literal;

import visiting.LiteralVisitor;
import visiting.value.StringValue;

public class StringLiteral extends Literal{
	
	private final StringValue value;
	
    public StringLiteral(String value){
    	this.value = new StringValue(value);
    }

    @Override
	public StringValue getValue() {
		return value;
	}
    
	@Override
	public <T, U> T accept(LiteralVisitor<T, U> visitor, U ctx) {
		return visitor.visit(this, ctx);
	}

	
}

File: Carly_Michael/jectacular-ql/src/app/domain/ast/ql/expressions/literals/string-literal.ts
import {Location} from '../../../location';
import {ExpressionType} from '../expression-type';
import {ExpressionVisitor} from '../../visitors/expression-visitor';
import {Literal} from './literal';
import {BooleanLiteral} from './boolean-literal';

export class StringLiteral extends Literal {
  constructor(public readonly value: string, readonly location: Location) {
    super(location);
  }

  getValue(): any {
    return this.value;
  }

  getType(): ExpressionType {
    return ExpressionType.STRING;
  }

  accept<T>(visitor: ExpressionVisitor<T>): T {
    return visitor.visitStringLiteral(this);
  }

  // arithmetic

  add(literal: Literal): Literal {
    return literal.addString(this);
  }

  addString(literal: StringLiteral): Literal {
    return new StringLiteral(this.value + literal.value, this.location);
  }

  // boolean ops

  equals(literal: Literal): Literal {
    return literal.equalsString(this);
  }

  equalsString(literal: StringLiteral): Literal {
    return new BooleanLiteral(this.value === literal.value, this.location);
  }

  notEquals(literal: Literal): Literal {
    return literal.notEqualsString(this);
  }

  notEqualsString(literal: StringLiteral): Literal {
    return new BooleanLiteral(this.value !== literal.value, this.location);
  }
}
